transaction/core/src/ring_signature/mlsag.rs

```diff
@@ -133,7 +135,7 @@ impl RingMLSAG {
     // * `rng` - Randomness.
     pub fn sign<CSPRNG: RngCore + CryptoRng>(
         message: &[u8; 32],
-        ring: &[(Address, Commitment)],
```

Comment:
`Address` and `Commitment` were both type aliases for RistrettoPoint. The aliases gave some readability, but they didn't actually provide type safety (for example, you could pass an Address to a function that takes a Commitment). This PR removes them in favor of more specific newtypes.